From a5a99b5549702c65ff69913c1ef8641c6b630b14 Mon Sep 17 00:00:00 2001
From: Christopher James Halse Rogers <christopher.halse.rogers@canonical.com>
Date: Sat, 20 Jul 2013 21:40:43 +1000
Subject: [PATCH 504/513] dri: Add fd parameter to __DRIbuffer

This allows platforms which want to do PRIME FD passing rather than
flink, such as Wayland and Mir, to do so without having to uselessly
allocate a world-visible name for internal Mesa use and sacrifice their
security level with that. Not even saying that nowadays names are
deprecated and any proper graphics stack should just use FDs.

Change-Id: If30dd4554381f14eef58a746bccd10f6cec7acb6
Signed-off-by: Christopher James Halse Rogers <raof@ubuntu.com>
Signed-off-by: Tomasz Figa <tfiga@google.com>
---
 include/GL/internal/dri_interface.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/GL/internal/dri_interface.h b/include/GL/internal/dri_interface.h
index 2612302..72ede8b 100644
--- a/include/GL/internal/dri_interface.h
+++ b/include/GL/internal/dri_interface.h
@@ -921,10 +921,12 @@ struct __DRIbufferRec {
     unsigned int pitch;
     unsigned int cpp;
     unsigned int flags;
+    int fd;            /**< Only available with DRI2_LOADER_VERSION >= 4, */
+                       /**< Only valid if name == 0 */
 };
 
 #define __DRI_DRI2_LOADER "DRI_DRI2Loader"
-#define __DRI_DRI2_LOADER_VERSION 3
+#define __DRI_DRI2_LOADER_VERSION 4
 struct __DRIdri2LoaderExtensionRec {
     __DRIextension base;
 
-- 
2.8.0.rc3.226.g39d4020

